Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.NewArrayExpression.cs

```diff
@@ -44,5 +45,27 @@ public ReadOnlyCollection<Expression> Expressions
         return default(ReadOnlyCollection<Expression>);
       }
     }
+
+#if NETFRAMEWORK_4_0 || SILVERLIGHT_4_0 || SILVERLIGHT_5_0
+    //
+    // Summary:
+    //     Creates a new expression that is like this one, but using the supplied children.
+    //     If all of the children are the same, it will return this expression.
+    //
+    // Parameters:
+    //   expressions:
+    //     The System.Linq.Expressions.NewArrayExpression.Expressions property of the result.
+    //
+    // Returns:
+    //     This expression if no children are changed or an expression with the updated
+    //     children.
+    [Pure]
+    public NewArrayExpression Update(IEnumerable<Expression> expressions)
+    {
+      Contract.Requires(expressions != null);
+      Contract.Ensures(Contract.Result<NewArrayExpression>() != null);
```

Comment:
`Contract.Requires(Type.GetElementType() != typeof(void))`

From this: http://referencesource.microsoft.com/#System.Core/Microsoft/Scripting/Ast/NewArrayExpression.cs,150